src/util/props.js

```diff
@@ -0,0 +1,46 @@
+
+import { warn } from './warn'
+
+export function resolveProps (route, component, config) {
+  switch (typeof config) {
+
+    case 'undefined':
+      return
+
+    case 'object':
+      return config
+
+    case 'function':
+      return config(route)
+
+    case 'boolean':
+      if (!config) {
+        return
+      }
+      if (!component.props) {
+        return
+      }
+      const props = {}
+      for (const prop in route.params) {
+        const value = route.params[prop]
+        if (hasProp(component, prop)) {
+          const propType = component.props[prop].type
```

Comment:
Not ideal, but the alternative was either not to support Number type, or pass it as an string and get a propType warnings and subtle bugs in `itemsById.find(item => item.id === id)`

I think routing to ID is a common use-case to allow it in the {props: true} scenario